transformer_engine/pytorch/attention.py

```diff
@@ -4362,15 +4364,13 @@ def forward(
         fp8_meta,
         deterministic,
     ):
+        is_input_fp8 = False
         if fp8:
             fused_attention_backend = FusedAttnBackend["FP8"]
             fp8_dtype_forward = get_fp8_te_dtype(fp8_meta["recipe"], fprop_tensor=True)
-            if fp8_meta["recipe"].fp8_mha:
-                assert (
-                    isinstance(q, Float8Tensor)
-                    and isinstance(k, Float8Tensor)
-                    and isinstance(v, Float8Tensor)
-                ), "q/k/v must be Float8Tensors for FP8 MHA."
+            assert type(q) == type(k) == type(v), "q, k, and v must have the same type."
```

Comment:
Linter recommends using `isinstance()` rather than `type()` for a type check. What should I do here?

Reply:
I think using `type` is cleaner, but better to go along with what Pylint wants.
```suggestion
            if not isinstance(k, q.__class__) or not isinstance(v, q.__class__):
                raise TypeError(
                    f"q ({q.__class__.__name__}), k ({k.__class__.__name__}), "
                    f"and v ({v.__class__.__name__}) do not have the same type"
                )
```